receiver/jaeger/trace_receiver_test.go

```diff
@@ -0,0 +1,234 @@
+	jr, err := jaegerreceiver.New(context.Background(), tchannelPort, collectorHTTPPort)
+	if err != nil {
+		t.Fatalf("Failed to create new Jaeger Receiver: %v", err)
+	}
+	defer jr.StopTraceReception(context.Background())
+	t.Log("Starting")
+
+	sink := new(concurrentSpanSink)
+	if err := jr.StartTraceReception(context.Background(), sink); err != nil {
+		t.Fatalf("StartTraceReception failed: %v", err)
+	}
+	t.Log("StartTraceReception")
+
+	now := time.Now().UTC()
+	nowPlus10min := now.Add(10 * time.Minute)
+	nowPlus10min2sec := now.Add(10 * time.Minute).Add(2 * time.Second)
+
+	// 2. Then with a "live application", send spans to the Jaeger exporter.
+	jexp, err := jaeger.NewExporter(jaeger.Options{
+		Process: jaeger.Process{
+			ServiceName: "issaTest",
+			Tags: []jaeger.Tag{
+				jaeger.BoolTag("bool", true),
+				jaeger.StringTag("string", "yes"),
+				jaeger.Int64Tag("int64", 1e7),
+			},
+		},
+		CollectorEndpoint: fmt.Sprintf("http://localhost:%d/api/traces", collectorHTTPPort),
+	})
+	if err != nil {
+		t.Fatalf("Failed to create the Jaeger OpenCensus exporter for the live application: %v", err)
+	}
+
+	// 3. Now finally send some spans
+	spandata := []*trace.SpanData{
+		{
+			SpanContext: trace.SpanContext{
+				TraceID: trace.TraceID{0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07, 0x08, 0x09, 0x0A, 0x0B, 0x0C, 0x0D, 0x0E, 0x0F, 0x80},
+				SpanID:  trace.SpanID{0xAF, 0xAE, 0xAD, 0xAC, 0xAB, 0xAA, 0xA9, 0xA8},
+			},
+			ParentSpanID: trace.SpanID{0x1F, 0x1E, 0x1D, 0x1C, 0x1B, 0x1A, 0x19, 0x18},
+			Name:         "DBSearch",
+			StartTime:    now,
+			EndTime:      nowPlus10min,
+			Status: trace.Status{
+				Code:    trace.StatusCodeNotFound,
+				Message: "Stale indices",
+			},
+			Links: []trace.Link{
+				{
+					TraceID: trace.TraceID{0xF1, 0xF2, 0xF3, 0xF4, 0xF5, 0xF6, 0xF7, 0xF8, 0xF9, 0xFA, 0xFB, 0xFC, 0xFD, 0xFE, 0xFF, 0x80},
+					SpanID:  trace.SpanID{0xCF, 0xCE, 0xCD, 0xCC, 0xCB, 0xCA, 0xC9, 0xC8},
+					Type:    trace.LinkTypeChild,
+				},
+			},
+		},
+		{
+			SpanContext: trace.SpanContext{
+				TraceID: trace.TraceID{0xF1, 0xF2, 0xF3, 0xF4, 0xF5, 0xF6, 0xF7, 0xF8, 0xF9, 0xFA, 0xFB, 0xFC, 0xFD, 0xFE, 0xFF, 0x80},
+				SpanID:  trace.SpanID{0xCF, 0xCE, 0xCD, 0xCC, 0xCB, 0xCA, 0xC9, 0xC8},
+			},
+			Name:      "ProxyFetch",
+			StartTime: nowPlus10min,
+			EndTime:   nowPlus10min2sec,
+			Status: trace.Status{
+				Code:    trace.StatusCodeInternal,
+				Message: "Frontend crash",
+			},
+			Links: []trace.Link{
+				{
+					TraceID: trace.TraceID{0xF1, 0xF2, 0xF3, 0xF4, 0xF5, 0xF6, 0xF7, 0xF8, 0xF9, 0xFA, 0xFB, 0xFC, 0xFD, 0xFE, 0xFF, 0x80},
+					SpanID:  trace.SpanID{0xAF, 0xAE, 0xAD, 0xAC, 0xAB, 0xAA, 0xA9, 0xA8},
+					Type:    trace.LinkTypeParent,
+				},
+			},
+		},
+	}
+
+	for _, sd := range spandata {
+		jexp.ExportSpan(sd)
+	}
+	jexp.Flush()
+
+	// Simulate and account for network latency but also the reception process on the server.
+	<-time.After(200 * time.Millisecond)
```

Comment:
@pjanotti  yes, the traces are accumulated in a slice. For us to be able to use opencensus-go/exporter/jaeger.Exporter with this test, we can't invasively add a channel unfortunately and in deed, using -race slows things down. I am thinking perhaps of doing a raw post but I'd prefer an actual integration like test such as this one instead of mocking transfers.

Reply:
Unfortunately, nothing that can be done at this moment and that sleep is just to "simulate" network latency if any.